pkg/registry/apis/provisioning/resources/repository.go

```diff
@@ -73,8 +77,21 @@ func (r *repositoryResourcesFactor) Client(ctx context.Context, repo repository.
 		return nil, fmt.Errorf("create parser: %w", err)
 	}
 
+	signatures := map[string]repository.CommitSignature{}
```

Comment:
having this inside the repo feels odd --- it is only for export where it makes sense to lookup all users rather than look up one at a time (or fill the user from the request context)

Reply:
yeah, it doesn't feel right.  The problem is that the repository itself it's ready to "commit" based on this signature if found in the lookup which comes from the object metadata. So currently, it's pretty entangled. I will open a follow up PR for this.